public/app/features/explore/utils/logsVolumeResponse.ts

```diff
@@ -0,0 +1,18 @@
+import { DataQueryError, DataQueryResponse } from '@grafana/data';
+
+// Currently we can only infer if an error response is a timeout or not.
+export function isTimeoutErrorResponse(response: DataQueryResponse | undefined): boolean {
+  if (!response) {
+    return false;
+  }
+  if (!response.error && !response.errors) {
+    return false;
+  }
+
+  const errors = response.error ? [response.error] : response.errors || [];
```

Comment:
```suggestion
  const errors = response.error ? [response.error] : response.errors;
```

Should also not be needed because we make sure `response.errors` is set in the if statement before.

Reply:
I wish TS was as smart as you
![imagen](https://user-images.githubusercontent.com/1069378/229073505-5ae2e7a4-0f7a-46cc-b2d0-ef5c9e99aabe.png)

Reply:
Interesting... then disregard this.

Reply:
Has happened to me a few times, it's very frustrating because we're clearly use that it's not undefined.